### PR TITLE
[5.x] Move sites out of config

### PR DESCRIPTION
**Pull request by jesseleite**
## What is changing...

The idea here is to move `sites` configuration out of the php config directory, and into `content/sites.yaml`, so that we can provide a nice GUI for managing your site/sites (since the `config` directory is meant to be edited by devs only)...

CleanShot 2024-04-04 at 19 00 11

If you need multi-site, you enable it in `config/statamic/system.php`...

CleanShot 2024-04-03 at 12 08 13

We've also streamlined the `php please multisite` command to do this for you...

CleanShot 2024-04-04 at 19 02 34

This allows you to manage multiple sites in the cp...

CleanShot 2024-03-14 at 16 48 35

Of course, you can still manage your sites by hand as before, but now that'll be in `content/sites.yaml`...

CleanShot 2024-03-13 at 23 42 21

## TODO...

- [x] Rip out `statamic.sites.sites` config
- [x] Add new `statamic.system.multisite` config
- [x] Update `Sites` class
    - [x] Deprecate old `setConfig()` method, with deprecated backwards compatibility fallback for addons
    - [x] Add replacement `setSites()` and `setSiteValue()` methods
    - [x] Finish massaging yaml in `save()` method
        - [x] Yaml should pretty much mirror what was in PHP `sites` config
        - [x] Remove explicit `direction` from config
            - [x] Related PR to help with this https://github.com/statamic/cms/pull/9730
    - [x] Resolve antlers for dynamic strings when getting sites
        - [x] Ensure raw/unresolved antlers is saved to yaml
- [x] Add new GUI to manage site/sites
    - [x] Stacked grid when multisite
        - [x] Related PR to help with this https://github.com/statamic/cms/pull/9733
	- [x] Warn user when they try to change site handle(s)
	- [x] Related PR to help with this https://github.com/statamic/cms/pull/9834
- [x] Refactor all the things that called `config('statamic.sites.sites')` to use new yaml sites config
- [x] Refactor `php please multisite` command
    - [x] Related PRs to help with this
        - [x] https://github.com/statamic/cms/pull/9763
        - [x] https://github.com/statamic/cms/pull/9778
    - [x] Update boolean in config/sites.php
    - [x] Update file/directory structure for multisite
    - [x] Tell them they can then manage new sites in CP
- [x] Update script
    - [x] To insert new `multisite` config into their `config/statamic/system.php`
    - [x] To move their sites from `config/statamic/sites.php` to `content/sites.yaml`
    - [x] To delete `config/statamic/sites.php` config, since it is no longer needed
- [x] Tests
	- [x] Update and pass all existing tests
	- [x] Add new test coverage on sites.yaml related stuff
	- [x] Test update script, since this is more critical feature for people
- [x] Documentation https://github.com/statamic/docs/pull/1307

Closes https://github.com/statamic/ideas/issues/1080

### CODE REVIEW

**Review by jasonvarga — CHANGES_REQUESTED** [review]

File: src/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -100,9 +100,7 @@ public function register()
             $this->mergeConfigFrom("{$this->root}/config/$config.php", "statamic.$config");
         });
 
-        $this->app->singleton(Sites::class, function () {
-            return new Sites(config('statamic.sites'));
-        });
+        $this->app->singleton(Sites::class, fn () => new Sites);
```
[the remark this answers is not in the document]
Comment (jesseleite) [reply src/Providers/AppServiceProvider.php]: Facades are singletons, so I think I can actually just remove that line altogether.

**Review by jasonvarga — APPROVED** [review]

**Comment by robdekort** [comment]
I'm wondering if this should be in the CP nav under `Content` by default. Perhaps it's better under `Tools` or `Utilities`. It's something you rarely need.

**Comment by jasonvarga** [comment]
That's a good thought.

**Comment by ajnsn** [comment]
With more complex page configurations, it is now much more difficult to achieve a reliable result with all these curly brackets as placeholders for URLs etc. - especially if you have also used PHP enums or collections/arrays in the "attributes".

Code analysis and testing also now seem more difficult to achieve.

Is there still a way e.g. to override this `sites.yaml` using PHP or is something like that planned? 

Thanks!

**Comment by duncanmcclean** [comment]
We're not planning on adding back a PHP method for configuring sites.

However, in your case, you could potentially override the sites in a service provider, like this:

```php
// app/Providers/AppServiceProvider.php

use Statamic\Facades\Site;
use Statamic\Statamic;

Statamic::booted(function () {
    Site::setSites([
        'default' => [
            'name' => 'Default',
            'locale' => 'en_US',
            'url' => YourEnum::Url,
        ],
    ]);
});
```

**Comment by ajnsn** [comment]
Thanks!

**Pull request merged**